src/hooks/useDisabledStandardizedVariables.ts

```diff
@@ -0,0 +1,35 @@
+import { useColumns, useStandardizedVariables } from '~/stores/FreshNewStore';
+
+/**
+ * Returns a Set of standardized variable labels that should be disabled in the dropdown.
+ * These are variables that:
+ * 1. Are already mapped to at least one column
+ * 2. Have can_have_multiple_columns = false (single-column only)
+ * 3. Are not multi-column measures
+ */
+export function useDisabledStandardizedVariables(): Set<string> {
+  const columns = useColumns();
+  const standardizedVariables = useStandardizedVariables();
+
+  const disabledLabels = new Set<string>();
+
+  Object.values(columns).forEach((column) => {
+    const variableId = column.standardizedVariable;
+
+    if (variableId) {
+      const variable = standardizedVariables[variableId];
+
+      if (variable) {
+        // Only include if it's a single-column variable (not multi-column measure and can't have multiple columns)
+        const isSingleColumn =
+          variable.is_multi_column_measure !== true && variable.can_have_multiple_columns === false;
+
+        if (isSingleColumn) {
+          disabledLabels.add(variable.name);
+        }
+      }
+    }
+  });
```

Comment:
alternatively to what I suggested above to make this hook return an object with the required shape for ColumnAnnotation, it may be enough to just return SV IDs here (not labels). Then a second hook can use that hook and also the `useStandardizedVariables` hook to create the expected shape for ColumnAnnotations. That would allow for each hook to remain focused on a single thing. I'd say the important part is to not pass labels, but instead pass IDs

Reply:
Makes sense, and we can revisit/refactor once there is a use case